MonoGame.Framework/Graphics/Shader/ShaderStage.cs

```diff
@@ -4,5 +4,8 @@ internal enum ShaderStage
     {
         Vertex,
         Pixel,
+        Hull,
+        Domain,
+        Geometry,
```

Comment:
We should document exactly what these are even with it being internal.  Will help when we port to new platforms and also when we move to make the Shader API public.